horovod/mxnet/mpi_ops.cc

```diff
@@ -297,7 +309,21 @@ extern "C" int horovod_mxnet_broadcast_async(NDArray* input, NDArray* output,
                            {output->var()}, FnProperty::kNormal, 0,
                            "HorovodBroadcast");
 #endif
-  return 0;
+  return handle;
+}
+
+extern "C" int horovod_mxnet_poll(int handle) {
+  return handle_manager.PollHandle(handle) ? 1 : 0;
+}
+
+extern "C" int horovod_mxnet_wait_and_clear(int handle) {
+  API_BEGIN();
+  while (!handle_manager.PollHandle(handle)) {
```

Comment:
I have concerns about this since it will introduce contention on the mutex.

Reply:
updated the sequence of `callback` and `markdone` as we discussed. So this should no longer introduce race condition. Please review again

Reply:
Thanks. I will review after you push your changes.